app/renderer/components/navigation/navigator.js

```diff
@@ -150,6 +152,172 @@ class Navigator extends React.Component {
     ipc.off(messages.SHORTCUT_ACTIVE_FRAME_FORWARD, this.onForward)
   }
 
+  // BEM Level: navigator__menuBarAndNavigationBar__navigationBarWrapper
+  get navigationBarWrapper () {
```

Comment:
@luixxiul I believe that not having any code in render is confusing. I do understand the intention of moving the logic out, because each function corresponds to part of the BEM path.  While maintaining the styles becomes easier, maintaining the code becomes more difficult.

For reference, I personally only try to move JSX/HTML rendering logic out of render into a function if one of these conditions is met:
- method has logic in it (looping, etc)
- method is re-usable. For example, `backButton` and `forwardButton` are almost identical... it may be possible to reduce this down to one method which takes params
- method makes code easier to test (and test cases are included) 

In these cases, depending on the complexity of the method (length, lots of if/else, etc), it may make sense create a component instead.

Back to this PR.  Moving only the `navigationBarWrapper()` function back into the `render()` could be a nice compromise because you can still see some components being rendered (and then the calls out).

Because the implications for this (pulling JSX/HTML out to methods) could be serious, I think we should discuss more